sched/semaphore/sem_recover.c

```diff
@@ -86,7 +86,8 @@ void nxsem_recover(FAR struct tcb_s *tcb)
   if (tcb->task_state == TSTATE_WAIT_SEM)
     {
       FAR sem_t *sem = tcb->waitobj;
-      DEBUGASSERT(sem != NULL && atomic_read(NXSEM_COUNT(sem)) < 0);
+
+      DEBUGASSERT(sem != NULL);
 
       /* Restore the correct priority of all threads that hold references
```

Comment:
In case of mutex, only 1 can hold a reference, the holder itself. Should the call to nxsem_canceled be omitted when dealing with mutexes ?

Reply:
There may be several tasks blocked on a mutex, and one of them can be deleted while holding the mutex. Again, keeping the existing functionality...

Reply:
Yes, many can wait, but only 1 can hold